oml/utils/audios.py

```diff
@@ -0,0 +1,127 @@
+import base64
+from io import BytesIO
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from PIL import Image
+from torch import FloatTensor
+
+from oml.const import BLACK, TColor
+
+
+def default_spec_repr_func(audio: FloatTensor) -> FloatTensor:
+    """
+    Generate a default spectral representation (log-scaled MelSpec) from an audio signal.
+    Used primarily for visualization.
+
+    Args:
+        audio: The input audio tensor.
+
+    Returns:
+        The spectral representation of the input audio tensor.
+    """
+    from torchaudio.transforms import MelSpectrogram
+
+    melspectrogram = MelSpectrogram(sample_rate=16000, n_fft=2048, hop_length=512, n_mels=128)
+    melspec = melspectrogram(audio)
+    log_melspec = torch.log1p(melspec).squeeze(0)
+    return log_melspec
+
+
+def _visualize_audio(
+    spec_repr: FloatTensor, color: TColor = BLACK, draw_bbox: bool = True, return_b64: bool = False
+) -> Union[np.ndarray, str]:
+    """
+    Internal function to visualize an audio spectrogram.
+
+    Args:
+        spec_repr: The spectrogram representation as a FloatTensor.
+        color: The color of the bounding box.
+        draw_bbox: Whether to draw a bounding box around the spectrogram.
+        return_b64: Whether to return the image as a base64 string.
+
+    Returns:
+        If return_b64 is False, returns the image as a NumPy array.
+        If return_b64 is True, returns the image as a base64 string.
+    """
+    fig, ax = plt.subplots(figsize=(2.56, 2.56), dpi=100)
+
+    # actual image
+    ax.imshow(spec_repr, aspect="auto", origin="lower")
+
+    # bbox and axes
+    if draw_bbox:
+        frame_thickness = 5
+        for axis in ["top", "bottom", "left", "right"]:
+            ax.spines[axis].set_linewidth(frame_thickness)
+            ax.spines[axis].set_edgecolor(
+                [c / 255 for c in color]
+            )  # TODO: change all colors from arrays to strings in vis functions
```

Comment:
not sure about this todo
could you explain?

Reply:
why cant we use just strings instead of `Tuple[int, int, int]`? Like "read"/"black"/etc? Only for more customisation?

Reply:
yes, don't want to overcomplicate it and support two types of inputs or maintain a mapping from strings to the values

Reply:
I mean that `ax.spines[axis].set_edgecolor()` accepts as input string values like `red`,`black`,etc. Why did you use tuples of ints in the first place?

Reply:
when I first introduced colors there was opencv
anyway, I think it's matter of taste 
so, I would rather remove the todo

Reply:
done